manifests/0000_90_kube-controller-manager-operator_04_servicemonitor-controller-manager.yaml

```diff
@@ -78,3 +78,46 @@ spec:
   selector:
     matchLabels:
       prometheus: kube-controller-manager
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    k8s-app: kube-controller-manager
+  name: cluster-policy-controller
+  namespace: openshift-kube-controller-manager
+  annotations:
+    include.release.openshift.io/ibm-cloud-managed: "true"
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+spec:
+  endpoints:
+    - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+      interval: 30s
+      metricRelabelings:
+        - action: drop
+          regex: etcd_(debugging|disk|request|server).*
+          sourceLabels:
+            - __name__
+        - action: drop
+          regex: rest_client_request_latency_seconds_(bucket|count|sum)
+          sourceLabels:
+            - __name__
+        - action: drop
+          regex: root_ca_cert_publisher_sync_duration_seconds_(bucket|count|sum)
+          sourceLabels:
+            - __name__
+      port: https
+      scheme: https
+      path: /cluster-policy-controller/metrics
```

Comment:
so, basically, you add a reverse proxy so you multiplex metrics by url

https://host:port/metrics -> kcm
https://host:port/cluster-policy/cluster-policy-controller/metrics -> cpc 

I'm not familiar with Openshift, is this the common way to do this in Openshift, is an interesting case indeed?

Reply:
Yes. But I am not sure how common (if anybody even does this approach) this is in openshift. 

We decided to use this approach after a discussion with installer people: apparently there is no process/solution to update (open/close) new ports on our nodes (usually via cloud provider API) and we do it on install only. And it seems there is no plan for such a solution in the future. This is not a problem for new clean installs, but we are unable to open a port for cluster-policy-controller (10357) on upgrades.

The proxy seems the most feasible solution how to achieve scrapping of our metrics.

Reply:
It was mostly curiosity ,  agree the proxy seems the simplest and easiest solution to this problem